### ISSUE TITLE
Use float64 instead of float32 in logger & metric & result

### ISSUE THREAD

**Issue by qqueing** [PyTorchLightning/pytorch-lightning#11386]
## 🐛 Bug

Torch metrics or logs use float32 as the default type.
However, this handling is not appropriate for large data sizes.

example in code, I log_dict - tensor(67128463., dtype=torch.float64)
but output result is tensor(67128464.) Maybe this logger use default type float32

https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/trainer/connectors/logger_connector/result.py#L217

I think the default type of log or metric would be float64.

torch.tensor(67128463., dtype=torch.float64)
Out[17]: tensor(67128463., dtype=torch.float64)
torch.tensor(67128463., dtype=torch.float32)
Out[18]: tensor(67128464.)

### To Reproduce

torch.tensor(67128463., dtype=torch.float64)
Out[17]: tensor(67128463., dtype=torch.float64)
torch.tensor(67128463., dtype=torch.float32)
Out[18]: tensor(67128464.)

In model, self.log_dict({"a": torch.tensor(67128463., dtype=torch.float64)})
output is tensor(67128464.)

### Environment

- PyTorch Lightning Version (e.g., 1.5.0): 1.5.4, but maybe all version..
- PyTorch Version (e.g., 1.10):
- Python version (e.g., 3.9):
- OS (e.g., Linux):
- CUDA/cuDNN version:
- GPU models and configuration:
- How you installed PyTorch (`conda`, `pip`, source):
- If compiling from source, the output of `torch.__config__.show()`:
- Any other relevant information:

### Additional context

cc @tchaton @rohitgr7 @carmocca @edward-io @ananthsub @kamil-kaczmarek @Raalsky @Blaizzy

**Comment by carmocca** [comment]
Hi @qqueing, thanks for your report.

Can you verify that the changes in https://github.com/PyTorchLightning/pytorch-lightning/pull/11401 fix your problem?

**Comment by qqueing** [comment]
> Hi @qqueing, thanks for your report.
> 
> Can you verify that the changes in #11401 fix your problem?

It is good. works fine
